src/ast.c

```diff
@@ -245,12 +245,26 @@ value_t fl_julia_scalar(fl_context_t *fl_ctx, value_t *args, uint32_t nargs)
     return fl_ctx->F;
 }
 
+value_t fl_julia_smallnum(fl_context_t *fl_ctx, value_t *args, uint32_t nargs)
+{
+    argcount(fl_ctx, "julia-smallnum?", nargs, 1);
+    if (isfixnum(args[0]) && numval(args[0]) < 32 && numval(args[0]) > -32)
+        return fl_ctx->T;
+    else if (iscvalue(args[0]) && fl_ctx->jl_sym == cv_type((cvalue_t*)ptr(args[0]))) {
+        jl_value_t *v = *(jl_value_t**)cptr(args[0]);
+        if (jl_is_long(v) && jl_unbox_long(v) < 32 && jl_unbox_long(v) > -32)
```

Comment:
The representation should be the same in that case. It would be an issue for splicing other integer types, though, but that's difficult to handle in general.